### PR TITLE
feat(music): implement Song Announce Feature

### PR DESCRIPTION
**Pull request by pryzmian**
- Extend metadata property in `ExtendedGuildQueuePlayerNode` type to include `lastMessage` property.
- Enable music bot to announce the start of a new song by sending a message to the channel.
- Add new event, `playerFinish`, to fetch and delete the last song-announce message if possible.
- Modify `play` command to accommodate these changes.

### CODE REVIEW

File: src/interactions/commands/player/play.ts [hunk 1]
```diff
@@ -189,14 +189,6 @@ class PlayCommand extends BaseSlashCommandInteraction {
                 const posistionFirstTrackInPlaylist = queue.tracks.data.length - searchResult.tracks.length + 1;
                 embedFooter = this.getDisplayFooterTrackPosition(posistionFirstTrackInPlaylist, translator);
             }
-        } else if (queue.currentTrack === track && queue.tracks.data.length === 0) {
```
Comment (mariusbegby) [inline src/interactions/commands/player/play.ts:192]: Same here, have this behind the config option mentioned in other comment.
```js
if(!enablePlayerStartMessages) {
    // send old message
}
```

**Comment by mariusbegby** [comment]
@pryzmian also eslint checks are failing, so please fix these formatting issues as well

**Comment by pryzmian** [comment]
Okay, I'll work on that

**Comment by pryzmian** [comment]
new config option done and formatting issues resolved

**Review by mariusbegby — APPROVED** [review]
Looks good! Well done 💯

**Pull request merged**